src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryDistributedCache.cs

```diff
@@ -75,13 +76,12 @@ public void Set(string key, byte[] value, DistributedCacheEntryOptions options)
             ThrowHelper.ThrowIfNull(value);
             ThrowHelper.ThrowIfNull(options);
 
-            var memoryCacheEntryOptions = new MemoryCacheEntryOptions();
-            memoryCacheEntryOptions.AbsoluteExpiration = options.AbsoluteExpiration;
-            memoryCacheEntryOptions.AbsoluteExpirationRelativeToNow = options.AbsoluteExpirationRelativeToNow;
-            memoryCacheEntryOptions.SlidingExpiration = options.SlidingExpiration;
-            memoryCacheEntryOptions.Size = value.Length;
-
-            _memCache.Set(key, value, memoryCacheEntryOptions);
+            using ICacheEntry entry = _memCache.CreateEntry(key);
+            entry.AbsoluteExpiration = options.AbsoluteExpiration;
+            entry.AbsoluteExpirationRelativeToNow = options.AbsoluteExpirationRelativeToNow;
+            entry.SlidingExpiration = options.SlidingExpiration;
+            entry.Size = value.Length;
+            entry.Value = value;
```

Comment:
should this use `entry.Dispose()` for parity with the change applied to `MemoryCacheExtensions` in this same PR?

Reply:
In practice `entry.Dispose()` would be slightly more efficient, but have to use `using` because someone could enable `TrackLinkedCacheEntries` for the `MemoryCache` options passed to this class.

Reply:
That raises complex questions about what the correct behaviour even *is* if an exception is thrown and we've only done half the work - could that result in us committing something incorrect?

Reply:
No, until `Value` is set, the entry is incomplete and `Dispose` won't store it in cache. `TrackLinkedCacheEntries` logic still needs to run however, because it needs to restore the state changed by entry allocation.